src/main/java/dao/FileReaderImpl.java

```diff
@@ -0,0 +1,34 @@
+package dao;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import model.FruitTransaction;
+import model.OperationType;
+
+public class FileReaderImpl implements dao.FileReader {
+
+    @Override
+    public List<FruitTransaction> readFile(String fileName) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
```

Comment:
It's good that you're using try-with-resources, but FileReader should be wrapped with InputStreamReader to specify the charset, e.g., new InputStreamReader(new FileInputStream(fileName), StandardCharsets.UTF_8).